repo/packages/K/kubernetes/3/package.json

```diff
@@ -11,7 +11,7 @@
     "icon-large": "https://downloads.mesosphere.com/kubernetes/artifacts/images/k8s-256x256.png"
   },
   "tags": ["mesosphere", "framework", "pod", "label", "paas"],
-  "preInstallNotes": "In order for Kubernetes to start successfully all resources must be available in the cluster including ports, CPU shares and RAM.\nWe recommend a minimum of 1 node with 1 CPU share and 1 GB of RAM available for use by the Kubernetes service.\nNote that the service is alpha and there may be bugs, including possible data loss, incomplete features, incorrect documentation or other discrepancies.",
+  "preInstallNotes": "In order for Kubernetes to start successfully all resources must be available in the cluster including ports, CPU shares and RAM.\nWe recommend a minimum of 1 node with 1 CPU share and 1 GB of RAM available for use by the Kubernetes service.\nNote that the service is alpha and there may be bugs, including possible data loss, incomplete features, incorrect documentation or other discrepancies.\n\nWARNING: The standard configuration launches an integrated etcd WITHOUT PERSISTENCE.\n         For production follow http://docs.mesosphere.com/services/kubernetes/#install\n         to configure kubernetes with a highly available etcd cluster.\n",
```

Comment:
persistence == save to disk. the way you've written it is confusing because
it etcd data logs are persistent, even in the embedded case. what's not
true in the embedded case is that etcd isn't highly available.

perhaps a different way to think about it is w/ respect to framework
failover. i wonder if it makes sense to drop the failover-timeout to
something much lower if the embedded etcd is used: w/ the embedded
container framework failover is meaningless and it benefits the user to
avoid dangling pods in this case. if an external etcd is used, then we keep
a nice long failover-timeout.


On Tue, Dec 8, 2015 at 10:59 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In repo/packages/K/kubernetes/3/package.json
> <https://github.com/mesosphere/multiverse/pull/58#discussion_r46971570>:
>
> > @@ -11,7 +11,7 @@
> >      "icon-large": "https://downloads.mesosphere.com/kubernetes/artifacts/images/k8s-256x256.png"
> >    },
> >    "tags": ["mesosphere", "framework", "pod", "label", "paas"],
> > -  "preInstallNotes": "In order for Kubernetes to start successfully all resources must be available in the cluster including ports, CPU shares and RAM.\nWe recommend a minimum of 1 node with 1 CPU share and 1 GB of RAM available for use by the Kubernetes service.\nNote that the service is alpha and there may be bugs, including possible data loss, incomplete features, incorrect documentation or other discrepancies.",
> > +  "preInstallNotes": "In order for Kubernetes to start successfully all resources must be available in the cluster including ports, CPU shares and RAM.\nWe recommend a minimum of 1 node with 1 CPU share and 1 GB of RAM available for use by the Kubernetes service.\nNote that the service is alpha and there may be bugs, including possible data loss, incomplete features, incorrect documentation or other discrepancies.\n\nWARNING: The standard configuration launches an integrated etcd WITHOUT PERSISTENCE.\n         For production follow http://docs.mesosphere.com/services/kubernetes/#install\n         to configure kubernetes with a highly available etcd cluster.\n",
>
> It's the PERSISTENCE which hurts the user most. A container restart on
> another node will even lead to another framework id and dangling pods. The
> cluster is essentially broken by then.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/multiverse/pull/58/files#r46971570>.
>